src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1184,11 +1184,24 @@ private void ExecuteAndCatchIfThrows_NoLock<TArg>(DiagnosticAnalyzer analyzer, A
                 cancellationToken.ThrowIfCancellationRequested();
                 analyze(argument);
             }
-            catch (Exception ex) when (HandleAnalyzerException(ex, analyzer, info, OnAnalyzerException, _analyzerExceptionFilter, cancellationToken))
+            catch (Exception ex) when (HandleAnalyzerException(analyzer, ex, info) &&
+                HandleAnalyzerException(ex, analyzer, info, OnAnalyzerException, _analyzerExceptionFilter, cancellationToken))
```

Comment:
Why do we need a second `HandleAnalyzer` function here? At a glance seems like the logic could just be rolled into the existing one. Doesn't seem like the other one is called anywhere else in the product (local search is acting up today though could be missing somtehnig)

Reply:
The existing one is used in more places:

https://github.com/dotnet/roslyn/blob/5b73439b70c5c5263b5acbb69b426c0264f14af5/src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs#L1214

https://github.com/dotnet/roslyn/blob/5b73439b70c5c5263b5acbb69b426c0264f14af5/src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs#L1226

I think it's not worth propagating the feature flag to those places as they only evaluate properties `SupportedDiagnostics` and `SupportedSuppressions` (and looks like propagating the feature flag there would be complex as that code has access only to `CompilationOptions`, not `Compilation`).